### PR TITLE
SOC-3392 | display who deleted a post

### PR DESCRIPTION
**Pull request by slayful**
## Links

* http://wikia-inc.atlassian.net/browse/SOC-3392

### CODE REVIEW

**Comment by slayful** [comment]
Will take care of the linter warnings, after working a little bit on the BE task which blocks this ticket.

File: front/main/app/components/discussion-post-card-base.js [hunk 1]
```diff
@@ -19,9 +19,9 @@ export default Component.extend(
 		shouldActivateLinks: Ember.computed.alias('isDetailsView'),
 		shouldTruncateContent: Ember.computed.not('isDetailsView'),
 
-		showTopNote: computed('isDeleted', 'isReported', 'post.isLocked', function () {
-			return !this.get('isDeleted') && this.get('isReported') || this.get('showRepliedTo') ||
-				this.get('post.isLocked');
+		showTopNote: computed('isDeleted', 'isReported', 'isLocked', function () {
```
Comment (bkoval) [inline front/main/app/components/discussion-post-card-base.js]: Let's add 'showReplied' to to the triggers here. Previous lack of it was probably a bug.

File: front/main/app/components/discussion-post-card-top-note.js [hunk 2]
```diff
@@ -37,15 +37,20 @@ export default Ember.Component.extend({
...
 	/**
 	 * Computes text for the post-card note
 	 */
-	topNoteText: Ember.computed('isReported', 'post.isLocked', 'post.reportDetails.count', function () {
-		// this block prepares 'reported posts' texts for moderators (regular user should never have post.reportDetails)
-		if (this.get('isReported') && this.get('canModerate') && this.get('post.reportDetails')) {
+	topNoteText: Ember.computed('isReported', 'post.isLocked', 'post.reportDetails.count',
+		'post.isDeleted', function () {
+		if (this.get('post.isDeleted') && this.get('canModerate')) {
+
```
Comment (bkoval) [inline front/main/app/components/discussion-post-card-top-note.js]: No need for this new line :)

File: front/main/app/components/discussion-post-card-top-note.js [hunk 3]
```diff
@@ -37,15 +37,20 @@ export default Ember.Component.extend({
...
+			return i18n.t('main.deleted-by', this.get('topNoteTextContext'));
+		} else if (this.get('isReported') && this.get('canModerate') && this.get('post.reportDetails')) {
+			// this block prepares 'reported posts' texts for moderators (regular user should never have post.reportDetails)
 			if (this.get('showRepliedTo')) {
```
Comment (bkoval) [inline front/main/app/components/discussion-post-card-top-note.js]: Do we need this empty line? :)

File: front/main/app/mixins/discussion-moderation-model.js [hunk 4]
```diff
@@ -66,10 +67,11 @@ export default Mixin.create({
 	deleteReply(reply) {
 		return request(M.getDiscussionServiceUrl(`/${this.wikiId}/posts/${reply.get('id')}/delete`), {
 			method: 'PUT'
-		}).then(() => {
+		}).then((response) => {
 			reply.setProperties({
 				isDeleted: true,
-				isReported: false
+				isReported: false,
+				lastDeletedBy: response.lastDeletedBy,
```
Comment (slayful) [inline front/main/app/mixins/discussion-moderation-model.js]: @bkoval these are the fixes to the issue found by @tortila

File: front/main/app/models/discussion/domain/post.js [hunk 5]
```diff
@@ -102,7 +103,7 @@ DiscussionPost.reopenClass({
 			creationTimestamp: DiscussionPost.getThreadDataTimestamp(threadData.creationDate),
 			isFollowed: threadData.isFollowed,
 			isLocked: !threadData.isEditable,
-			lastEditedBy: DiscussionContributor.create(threadData.lastEditedBy),
+			// lastEditedBy: DiscussionContributor.create(threadData.lastEditedBy),
```
Comment (bkoval) [inline front/main/app/models/discussion/domain/post.js]: Dev code leftovers?

File: front/main/app/components/discussion-post-card-top-note.js [hunk 6]
```diff
@@ -43,48 +43,46 @@ export default Ember.Component.extend({
...
-			} else if (!this.get('isReply')) {
-				if (this.get('post.isLocked')) {
-					return i18n.t('main.reported-by-and-locked', this.get('topNoteTextContext'));
+	topNoteText: Ember.computed('isReported', 'post.isLocked', 'post.reportDetails.count',
+		'post.isDeleted', 'post.lastDeletedBy.name', function () {
+			if (this.get('post.isDeleted') && this.get('canModerate')) {
+				let username = this.get('post.lastDeletedBy.name') || '';
+				return i18n.t('main.deleted-by', {userName: username, ns: 'discussion'});
+			} else if (this.get('isReported') && this.get('canModerate') && this.get('post.reportDetails')) {
+				// this block prepares 'reported posts' texts for moderators (regular user should never have post.reportDetails)
```
Comment (bkoval) [inline front/main/app/components/discussion-post-card-top-note.js]: Please, either consider refactoring this code or create a refactoring ticket and link it here. There is no point in extending this logic in it's current state as it's not maintainable anymore.

Comment (slayful) [inline front/main/app/components/discussion-post-card-top-note.js]: https://wikia-inc.atlassian.net/browse/SOC-3848

**Pull request merged**